zipline/data/data_portal.py

```diff
@@ -696,6 +766,25 @@ def get_history_window(self, assets, end_dt, bar_count, frequency, field,
 
         # forward-fill if needed
         if field == "price" and ffill:
+            assets_with_nan_index = np.where(pd.isnull(df.iloc[0, :]))[0]
+            assets_with_leading_nan = [
+                assets[i] for i in assets_with_nan_index]
+            for assets in assets_with_leading_nan:
+                if frequency == "1m":
+                    data_frequency = 'minute'
+                elif frequency == "1d":
+                    data_frequency = 'daily'
+                else:
+                    raise Exception(
+                        "Only 1d and 1m are supported for forward-filling.")
+                dt_to_fill = df.index[0]
+                perspective_dt = df.index[-1]
+                previous_dt = self.get_last_traded_dt(
+                    asset, dt_to_fill, data_frequency)
+                previous_value = self._get_adjusted_value(
+                    asset, field, previous_dt, perspective_dt,
+                    data_frequency)
+                df.loc[dt_to_fill, asset] = previous_value
```

Comment:
What do suggest instead, `df[asset][0]`? (As a force of habit, I've been use `loc` etc. when using two-coordinates, but not as important in this case where only one item is being written.)